target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_simple.c

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
```

Comment:
Any plans to upstream this?

Reply:
This is a mtdsplit parser of OpenWrt, for splitting firmware partition to kernel and rootfs.

Reply:
That is possible, but I dont think that we need to have the entirely new file in this repository as there are not enough people, who would maintain it and keep it updated. 

- Are there any plans that other routers will use it as well?
- Why it was necessary to have the simple variant instead of using existing variants?

Reply:
> Are there any plans that other routers will use it as well?

I don't know. May or may not use.

> Why it was necessary to have the simple variant instead of using existing variants?

Unless there is a device that I am not aware of, there is no other device that uses the same header, and the existing mtdsplit parsers cannot handle it.
At first, I created a mtdsplit parser exclusively for the header format of this device, but since there was a possibility that this device would be the only model to be used in the future, I changed it to a general-purpose specification.